### PR TITLE
Added Avalanche-CLI troubleshooting doc

### PR DESCRIPTION
**Pull request by dasconnor**
Add doc to address common debugging concerns Avalanche-CLI problems.

### CODE REVIEW

File: docs/subnets/troubleshoot-deployments.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+avalanche network clean --hard
+```
+
+Next, look for any lingering AvalancheGo processes with:
+
+```shell
+ps aux | grep avalanchego
+```
+
+If any processes are running, you need to stop them before you can launch your VM with Avalanche-CLI.
```
Comment (felipemadero) [inline docs/subnets/troubleshoot-deployments.md:36]: eg killall avalanchego

File: docs/subnets/troubleshoot-deployments.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+:::warning
+
+If you're running a validator node on the same box you're using Avalanche-CLI, **don't** end any
+of these lingering AvalancheGo processes. This may shut down your validator and could affect
+your validation uptime.
```
Comment (felipemadero) [inline docs/subnets/troubleshoot-deployments.md:42]: probably highly recommend to use avalanche-cli on other machine

File: docs/subnets/troubleshoot-deployments.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+protocol specification shared by both AvalancheGo and the VM. Both components **must** be running
+the same RPC version for VM deployment to work. You can view [AvalancheGo's RPC compatibility broken
+down by release
+version](https://github.com/ava-labs/avalanchego/blob/master/version/compatibility.json).
+
+Your custom VM's RPC version is set by the version of AvalancheGo that you import. By default,
+Avalanche-CLI creates local Avalalanche networks that run the latest AvalancheGo release.
+
+Here's an example with real numbers from the AvalancheGo compatibility page. If the latest
+AvalancheGo release is version v1.9.4, then Avalanche-CLI deploys a network with RPC version 21. For
```
Comment (felipemadero) [inline docs/subnets/troubleshoot-deployments.md:77]: probably confusing 21 with 20 in `with RPC version 21`

File: docs/subnets/troubleshoot-deployments.md (hunk 1 above)
Comment (holisticode) [reply docs/subnets/troubleshoot-deployments.md:36]: Very dangerous for things like line 40, if there are other avago running on the same machine

**Review by holisticode — COMMENTED** [review]
Do we want to add the case of this morning? Like presence of manually installed plugins in the plugin directory?

File: docs/subnets/troubleshoot-deployments.md (hunk 1 above)
Comment (dasconnor) [reply docs/subnets/troubleshoot-deployments.md:36]: I agree with Fabio, I think that's too dangerous. I think it's safe that developers will know how to kill a process.

File: docs/subnets/troubleshoot-deployments.md (hunk 2 above)
Comment (dasconnor) [reply docs/subnets/troubleshoot-deployments.md:42]: added

File: docs/subnets/troubleshoot-deployments.md (hunk 3 above)
Comment (dasconnor) [reply docs/subnets/troubleshoot-deployments.md]: fixed

**Review by felipemadero — APPROVED** [review]

**Review by yulin-dong — APPROVED** [review]

**Pull request merged**